client2/src/game/Game.js

```diff
@@ -548,35 +575,12 @@ class Game extends Phaser.Scene {
         this.keyboardKeys.d.on("up", this.moveRightReleased, this);
 
         this.keyboardKeys.enterChat.on("down", () => {
-            if (this.player.hitPoints <= 0) {
-                // Close the box. Can't chat while dead.
-                this.GUI.chatInput.isActive = false;
-                this.GUI.chatInput.style.visibility = "hidden";
-                this.GUI.chatInput.blur();
-                this.GUI.chatInput.value = "";
-                return;
-            }
-            // Check if the chat input box is open.
-            if (this.GUI.chatInput.isActive === true) {
-                // Close the box, and submit the message.
-                this.GUI.chatInput.isActive = false;
-                this.GUI.chatInput.style.visibility = "hidden";
-                this.GUI.chatInput.blur();
-
-                // Don't bother sending empty messages.
-                if (this.GUI.chatInput.value !== "") {
-                    // Send the message to the server.
-                    window.ws.sendEvent("chat", this.GUI.chatInput.value);
-
-                    // Empty the contents ready for the next chat.
-                    this.GUI.chatInput.value = "";
-                }
-            }
-            else { // Not open, so open it.
-                this.GUI.chatInput.isActive = true;
-                this.GUI.chatInput.style.visibility = "visible";
-                this.GUI.chatInput.focus();
-            }
+            // Send player hitPoints and websocket to be used by chatInput
+            // player is undefined 'cause it's not defined,
```

Comment:
~~player~~hitPoints is undefined 'cause it's not defined,

Reply:
All of the player data stuff in Game.js became a mess to manage, so it was moved to the following state files, and the rest will be moving there too
Player state: https://github.com/Arcanorum/dungeonz/blob/feat/react/client2/src/shared/state/Player.js
or
Inventory state: https://github.com/Arcanorum/dungeonz/blob/feat/react/client2/src/shared/state/Inventory.js

Reply:
Removed at https://github.com/Arcanorum/dungeonz/pull/99/commits/559e46b0dab5d1a050e33f2b0faa246d76354a20

Reply:
Oh I saw those state files, it didn't cross my mind to use it.
I though player object was undefined on purpose for now.

Will definitely take a look at those state folder and see if I can implement things cleaner.

Reply:
fixed at 836dbba